app/repositories/application/GeneralApplicationRepository.scala

```diff
@@ -539,6 +552,21 @@ class GeneralApplicationMongoRepository(timeZoneService: TimeZoneService,
           "testGroups" -> "")))
         bsonCollection.findAndModify(query, updateOp).map(_.result[Candidate])
       }
+      case AddMissingPhase2ResultReceived =>
+        val query = BSONDocument("$and" -> BSONArray(
+          BSONDocument("applicationId" -> application.applicationId),
+          BSONDocument("applicationStatus" -> ApplicationStatus.PHASE2_TESTS),
+          BSONDocument(s"progress-status.${ProgressStatuses.PHASE2_TESTS_RESULTS_READY}" -> true),
+          BSONDocument(s"progress-status.${ProgressStatuses.PHASE2_TESTS_RESULTS_RECEIVED}" -> BSONDocument("$ne" -> true))
```

Comment:
We're re-executing part of the select query we used above to get the applications to modify.  If we're just doing this to make sure the application document hasn't changed in the meantime, should we re-run the whole query?